### ISSUE TITLE
Java: Issue with dotted annotations

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#8884]
**Describe the bug**
Processing Java code
```
public class Test {
    @java.io.Serial
    public void blah() {}
}
```
results in the following xml snippet:
```
  <memberdef kind="function" id="classTest_1a1b46ce04e863829723c4f62bc1ea8d43" prot="public" static="no" const="no" explicit="no" inline="yes" virt="non-virtual">
    <type>.io.Serial void</type>
    <definition>.io.Serial void Test.blah</definition>
    <argsstring>()</argsstring>
    <name>blah</name>
    <briefdescription>
    </briefdescription>
    ...
  </memberdef>
```
Note the trashy ``<type>.io.Serial void</type>``. Doxygen does not seem to strip away Java annotations reliably if there are dots in the name.

**Expected behavior**
I expect that the resulting code snippet would be
```
  <memberdef kind="function" id="classTest_1a1b46ce04e863829723c4f62bc1ea8d43" prot="public" static="no" const="no" explicit="no" inline="yes" virt="non-virtual">
    <type>void</type>
    <definition>void Test.blah</definition>
    <argsstring>()</argsstring>
    <name>blah</name>
    <briefdescription>
    </briefdescription>
    ...
  </memberdef>
```

**To Reproduce**
Find a Java file and doxygen config attached. The issue is demonstrated like mentioned above.
After running doxygen on the example, the files in the subfolder ``xml`` contain the snippet shown above.
[issue_Java_dotted_annotations.zip](https://github.com/doxygen/doxygen/files/7500022/issue_Java_dotted_annotations.zip)

**Version**
Tested with doxygen clone from 2021-11-08:
```
/blah/doxygen/build/bin/doxygen -v
1.9.3 (096a1c10e1e0b86edc8c561f49b5fd7881776ba7)
```

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
TBH I did not check for duplicates. Please excuse if this is a known issue.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8887

**Comment by Stewori** [comment]
Thanks! Will check it out.

**Comment by Stewori** [comment]
I can confirm that your PR fixes this issue, at least for my setting.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
